bin/report

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+# bin/report <build-dir> <cache-dir> <env-dir>
+
+### Configure environment
+
+set -o errexit    # always exit on error
+set -o pipefail   # don't ignore exit codes when piping output
+
+BUILD_DIR=${1:-}
+
+packages=()
+custom_packages=()
+custom_repositories=()
+
+while IFS= read -r line; do
+  if grep --silent --invert-match -e "^#" -e "^\s*$" <<< "${line}"; then
```

Comment:
So, this level of indentation could still be removed if you moved the `grep` to the bottom `done`, feeding the whole `while` loop only the lines that actually matter.

This is also much more efficient because it's a single pipeline and a single invocation of `grep`, instead of a `grep` call for every line.